lib/utils/device/api.ts

```diff
@@ -256,24 +261,35 @@ export class DeviceAPI {
 			}
 		}
 
-		return new Promise((resolve, reject) => {
-			return request(opts, (err, response, body) => {
-				if (err) {
-					return reject(err);
-				}
-				switch (response.statusCode) {
-					case 200:
-						return resolve(body);
-					case 400:
-						return reject(new ApiErrors.BadRequestDeviceAPIError(body.message));
-					case 503:
-						return reject(
-							new ApiErrors.ServiceUnavailableAPIError(body.message),
-						);
-					default:
-						return reject(new ApiErrors.DeviceAPIError(body.message));
-				}
+		const doRequest = async () => {
+			return await new Promise((resolve, reject) => {
+				return request(opts, (err, response, body) => {
+					if (err) {
+						return reject(err);
+					}
+					switch (response.statusCode) {
+						case 200:
+							return resolve(body);
+						case 400:
+							return reject(
+								new ApiErrors.BadRequestDeviceAPIError(body.message),
+							);
+						case 503:
+							return reject(
+								new ApiErrors.ServiceUnavailableAPIError(body.message),
+							);
+						default:
+							return reject(new ApiErrors.DeviceAPIError(body.message));
+					}
+				});
 			});
+		};
+
+		return await retry({
+			func: doRequest,
+			initialDelayMs: 2000,
+			maxAttempts: 6,
+			label: `Supervisor API (${opts.method} ${(opts as any).url})`,
```

Comment:
@20k-ultra, @cywang117, this "livepush" CLI PR proposes to retry failed requests to the supervisor API. For GET requests it should be fine, but it could also include a POST request to `v2/local/target-state`. I don't see a problem with retrying such requests if they fail, do you? The motivation is the following paid support thread: https://jel.ly.fish/support-thread-1-0-0-front-cnv-b4ord71

Reply:
The POST retries to /v2/local/target-state should be fine too @pdcastro.

I can't tell from the code changes, but I saw that on sentry, the CLI logs the status code of each Supervisor API response. Would it also log retries and errors?

Reply:
Each retry would _not_ be logged on Sentry, but when/if `maxAttempts` is reached, the error of the last retry would be logged on Sentry. Is this OK?

Reply:
That sounds good to me!